datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/BatchAddTermsResolver.java

```diff
@@ -39,6 +46,15 @@ public CompletableFuture<Boolean> get(DataFetchingEnvironment environment) throw
 
       // First, validate the batch
       validateTerms(termUrns);
+
+      // if we're adding terms to one sub-resource (ie. schema field) attempt to add to siblings if it fails
+      if (resources.size() == 1 && resources.get(0).getSubResource() != null) {
```

Comment:
this type of feature-specific server-side hack is not ideal.

Can we break this logic into a separate method? called "handleAddSingleTermToSchemaField" 

then in the java docs we can explain the intricacies with relation to siblings feature

Reply:
(same with above resolver)

Reply:
totally agreed. moving this to a separate method with a better comment explaining what's going on here